### PR TITLE
CIFAR 94.03%

### PR DESCRIPTION
**Pull request by g1y5x3**
Was using CIFAR to practice disk tensor to improve image loading speed for maskrcnn since its easier to debug but then decided to give it a shot. The new dataloader now returns raw format (uint8 for images and int64 for labels) but as disk tensor. All the preprocessing are done inside GPU now (/255, normalize, reshape, flip LR). Added `whitening` (still done inside torch and will remove) and `cutmix`. 

I think 94% should be just a hyper-parameter search away but we will see ([search I did from last night](https://api.wandb.ai/links/g1y5x3/883u4bdb))

```
TTA=1 GPU=1 CUDA=0 BS=512 STEPS=1000 python examples/hlb_cifar10.py

999  322.51 ms run,    3.00 ms python,  319.52 ms CL,    0.99 loss, 0.000047 LR, 10.17 GB used,   3624.74 GFLOPS
eval 9249/10000 92.49%,    0.59 val_loss STEP=1000
```
Also, whitening adds additional kernels which caused test failure.

### CODE REVIEW

**Comment by g1y5x3** [comment]
### Current differences

- Preprocessing
    - [x]  pad reflect to 36x36
    - [x]  random crop back to 32x32
    - [x]  random flip lr w/ p=0.5
    - [x]  cutmix 3x3 patches randomly
- Model
    - [ ]  weight normalization during forward pass
    - [ ]  [[TemperatureScaler](https://github.com/tysam-code/hlb-CIFAR10/blob/3bb104ce16d169e14783c406a3ac9b5a1e5f6678/main.py#L220-L226)](https://github.com/tysam-code/hlb-CIFAR10/blob/3bb104ce16d169e14783c406a3ac9b5a1e5f6678/main.py#L220-L226) (literally just a scaler)
    - [ ]  [[loss scaling](https://github.com/tysam-code/hlb-CIFAR10/blob/3bb104ce16d169e14783c406a3ac9b5a1e5f6678/main.py#L591-L594)](https://github.com/tysam-code/hlb-CIFAR10/blob/3bb104ce16d169e14783c406a3ac9b5a1e5f6678/main.py#L591-L594)
- Optimization
    - [x]  separate learning rate for the weights and bias
    - [ ]  tuning hyper-parameters (17 digits after decimal point is pretty much unnecessary)
- Others
    - [ ]  send to `channels_last` for improving tensor core occupancy (mostly for speed)

**Comment by g1y5x3** [comment]
Currently sitting at **94.03%**. Nice thing to know is that almost all hyper-parameters are the same as the [original repository](https://github.com/tysam-code/hlb-CIFAR10/blob/3bb104ce16d169e14783c406a3ac9b5a1e5f6678/main.py#L48-L69). The only thing I tuned was [`final_lr_ratio`](https://github.com/tysam-code/hlb-CIFAR10/blob/3bb104ce16d169e14783c406a3ac9b5a1e5f6678/main.py#L563).
Also, able to reduce the step time from `320 ms` to `193 ms` and saved memory from `10.17 GB` to `5.99 GB`
```
GPU=1 python examples/hlb_cifar10.py

999  192.95 ms run,    2.75 ms python,  190.20 ms CL,  473.66 loss, 0.000070 LR, 5.99 GB used,   3494.46 GFLOPS
eval 9403/10000 94.03%,    0.40 val_loss STEP=1000
```
